servatrice/src/servatrice_database_interface.cpp

```diff
@@ -111,11 +111,52 @@ bool Servatrice_DatabaseInterface::usernameIsValid(const QString &user)
     return (result.size() > 0);
 }
 
+// TODO move this to Server
 bool Servatrice_DatabaseInterface::getRequireRegistration()
 {
     return settingsCache->value("authentication/regonly", 0).toBool();
 }
 
+bool Servatrice_DatabaseInterface::registerUser(const QString &userName, const QString &realName, ServerInfo_User_Gender const &gender, const QString &passwordSha512, const QString &emailAddress, const QString &country, bool active)
+{
+    if (!checkSql())
+        return false;
+
+    QSqlQuery *query = prepareQuery("insert into {prefix}_users "
+            "(name, realname, gender, password_sha512, email, country, registrationDate, active) "
+            "values "
+            "(:userName, :realName, :gender, :password_sha512, :email, :country, UTC_TIMESTAMP(), :active)");
+    query->bindValue(":userName", userName);
+    query->bindValue(":realName", realName);
+    query->bindValue(":gender", getGenderChar(gender));
+    query->bindValue(":password_sha512", passwordSha512);
```

Comment:
I should probably have some validation that this is correct, but if they send junk it means that probably they just won't be able to connect